File: Jim/src/sk/fiit/jim/garbage/plan/PlanTournamentFreeRide.java
package sk.fiit.jim.garbage.plan;

import sk.fiit.jim.agent.AgentInfo;
import sk.fiit.jim.agent.highskill.*;
import sk.fiit.jim.agent.models.AgentModel;
import sk.fiit.jim.agent.models.EnvironmentModel;
import sk.fiit.robocup.library.geometry.Vector3D;

/**
 * Rewrited PlanTournamentFreeRide from ruby
 * 
 * @author Matej Badal
 */

public class PlanTournamentFreeRide extends Plan {

	public static PlanTournamentFreeRide getInstance() {
		return new PlanTournamentFreeRide();
	}

	public void replan() {
        AgentModel agentModel = AgentModel.getInstance();
           
		if (EnvironmentModel.beamablePlayMode()) {
			AgentInfo.logState("Beam");
			Vector3D start_position = Vector3D.cartesian(-5, 1, 0.4);
			Highskill_Queue.add(new Beam(start_position));
			this.beamed = true;
		} else if (agentModel.isOnGround() || agentModel.isLyingOnBack() || agentModel.isLyingOnBelly()) {
			AgentInfo.logState("GetUp");
			Highskill_Queue.add(new GetUp());
		} else {
			AgentInfo.logState("FreeRide");
			Highskill_Queue.add(new FreeRide());
		}
	}
}

File: Old_xml_and_scripts/Jim/plan/old-predUcesanim/planTournamentFreeRide.rb
### planTournamentFreeRide.rb ###
#	@author: A55Kickers			#
#################################

class PlanTournamentFreeRide < Plan
  
  def replan
    @plan.clear
    @agentInfo.whereIsBall();
    @agentInfo.isBallMine();
    @agentInfo.nearBall();
    @agentInfo.ballControlPosition();
    @agentInfo = Java::sk.fiit.jim.agent.AgentInfo.getInstance
	
    if EnvironmentModel.beamablePlayMode
    	@agentInfo.loguj('Beam');
    	@plan << Beam.new(Java::sk.fiit.robocup.library.geometry.Vector3D.cartesian(-2, 0, 0.4))
    	@beamed = true
    elsif (me.on_ground? or me.is_lying_on_back? or me.is_lying_on_belly?)
    	@agentInfo.loguj('GetUp');
     	@plan << GetUp.new(Proc.new{me.on_ground?})
   	else
    	@agentInfo.loguj('Free Ride');
    	#@plan << LowSkill.new("free_ride") #TODO: sem doplnit low skill na turnajovu volnu jazdu
    end
  end
end